exporter/prometheusremotewriteexporter/helper.go

```diff
@@ -169,7 +182,11 @@ func getPromMetricName(desc *otlp.MetricDescriptor, ns string) string {
 	if b.Len() > 0 {
 		b.WriteString(delimeter)
 	}
-	b.WriteString(desc.GetName())
+	name := metric.GetName()
+	b.WriteString(name)
+
+	// do not add the total suffix if the metric name already ends in "total"
+	isCounter = isCounter && name[len(name)-len(totalStr):] != totalStr
```

Comment:
Maybe it's because I can only see the edits and not the full function, but I don't really understand this line. It looks like it's setting a boolean, but the comment doesn't match up.